src/Sulu/Bundle/AdminBundle/Resources/js/containers/Datagrid/Datagrid.js

```diff
@@ -1,53 +1,64 @@
 // @flow
 import {observer} from 'mobx-react';
+import {observable, action} from 'mobx';
 import React from 'react';
 import equal from 'fast-deep-equal';
 import Loader from '../../components/Loader';
 import PaginationDecorator from './PaginationDecorator';
 import DatagridStore from './stores/DatagridStore';
 import datagridAdapterRegistry from './registries/DatagridAdapterRegistry';
-import type {AdapterConfig} from './types';
-
-const INFINITE_SCROLL_TYPE = 'infiniteScroll';
+import AbstractAdapter from './adapters/AbstractAdapter';
+import AdapterSwitch from './AdapterSwitch';
 
 type Props = {
     onItemClick?: (itemId: string | number) => void,
     store: DatagridStore,
-    views: Array<string>,
+    adapters: Array<string>,
 };
 
+type CurrentAdapter = {
```

Comment:
Isn't just the key enough to keep track of the current adapter? With the key you could get the right adapter from the `datagridAdapterRegistry` anyway.